### DISCUSSION TITLE
Listing all failed projects?

### DISCUSSION THREAD

**Discussion by tstackhouse** [nrwl/nx#11932] [General]
I'm working on migrating to Nx 14, and with the new output format and the removal of `--only-failed`, I don't have a way to see all the failed tasks without rerunning my entire test suite from time to time as I'm working through fixing the failures from the upgrade.  Combined with the issues from jest/angular esm slowing down the test running it's really slowed down my workflow.

Is there a way to use `print-affected` or something to list all the failed projects?  Having a list of 5 followed by `...and XX more...` after I run my tests isn't the most helpful.

**Comment by hewelt** [comment]
This would be useful in our project too: we have a CI workflow where after running a `test` target, we'd like to upload artifacts only for projects where the task failed.

**Reply by cuddlecake** [reply]
Another use case: Making major changes across the project (upgrades for example) and running all tests. It's not a good experience having to rerun all the tests, even when you know that there are XX more that have failed, but are not getting listed.

**Comment by kopach** [comment]
The variable controlling how many Failed projects to print is located here https://github.com/nrwl/nx/blob/529ab9455ccb100e4518cd3884e284e7409629e9/packages/nx/src/tasks-runner/life-cycles/dynamic-run-many-terminal-output-life-cycle.ts#L381C13-L381C29
So, as a **workaround** what I did was to edit this file under `node_modules`
1. `code node_modules/nx/src/tasks-runner/life-cycles/dynamic-run-many-terminal-output-life-cycle.js`
2. search for `const numFailedToPrint` and change value to whatever you like
3. re-run tasks
Not ideal, but does the job. It's possible to later copy&paste/redirect output and work with it

**Comment by rujorgensen** [comment]
Is there a solution to this yet 🙂? I'm migrating to eslint flat config, and I have to re-run my complete linting suite every time I've fixed five libraries out of hundreds of failing targets. It takes roughly 25 minutes every time, I've been at it for days ☠️.

**Comment by msanchezc** [comment]
Is there any way to get the output of `nx affected` "Failed tasks" section in a list or JSON? I think that's a fairly common requirement for CI/CD.

**Comment by jiri-prokop-pb** [comment]
Would be amazing (and quite simple to implement, I think) to have possibility to set `numFailedToPrint` (mentioned in https://github.com/nrwl/nx/discussions/11932#discussioncomment-10760478) through env variable.

**Comment by AdamMcCormickOMF** [comment]
I know this is a bit of a necro-post, but this irked me so I wrote this: https://github.com/nrwl/nx/pull/34115
